cli-commands/flattener/flatten.js

```diff
@@ -2,27 +2,41 @@
 const fs = require('fs-extra');
 const path = require('path');
 let Profiler = require("../compiler/etherlime-compile/profiler");
-let {resolver, supplier} = require('./config.js')
+let { resolver, supplier } = require('./config.js')
 
 const versionRegex = /^\s*pragma\ssolidity\s+(.*?)\s*;/; //regex for pragma solidity version 
 const importRegex = /^\s*import(\s+).*$/gm; //regex for imported files
 
-const run = async (file, solcVersion) => {
-	if(solcVersion) {
+const returnFilesAndPaths = async (file, solcVersion) => {
+	if (solcVersion) {
 		supplier.config.version = solcVersion
 	}
 
+	let resolvedFiles = await resolveSources(`./contracts/${file}`)
+	let resolvedPaths = resolvePaths(resolvedFiles)
+	let orderedPaths = orderPaths(resolvedFiles, resolvedPaths)
+	return { resolvedFiles, orderedPaths }
+}
+
+const run = async (file, solcVersion) => {
+
 	try {
-		let resolvedFiles = await resolveSources(`./contracts/${file}`)
-		let resolvedPaths = resolvePaths(resolvedFiles)
-		let orderedPaths = orderPaths(resolvedFiles, resolvedPaths)
+		const { resolvedFiles, orderedPaths } = await returnFilesAndPaths(file, solcVersion);
 		recordFiles(file, resolvedFiles, orderedPaths)
 		console.log('Contract was flattened successfully. Check your "./flat" folder')
 	} catch (e) {
 		throw new Error(e.message);
 	}
-	
+};
 
+const runWithoutWriteFiles = async (file, solcVersion) => {
+
+	try {
+		const { resolvedFiles, orderedPaths } = await returnFilesAndPaths(file, solcVersion);
+		return returnFiles(file, resolvedFiles, orderedPaths)
+	} catch (e) {
```

Comment:
Is this really needed. Can we just let everything throw the error instead of creating new error that has the same message